pkg/querier/frontend/frontend.go

```diff
@@ -316,6 +316,9 @@ func (f *Frontend) Process(server Frontend_ProcessServer) error {
 		}
 
 		originalCtx := request.originalCtx
+		if request.request != nil && request.request.QueryRangeRequest != nil {
```

Comment:
Under what circumstances will request.request be nil?  I can't think of any.

Reply:
Idk it crashed in the unit tests and I couldn’t be bothered figuring out which one was nil.

Reply:
I removed `request.request != nil &&` and the tests passed...

Either way this is in the wrong place; if you look at L172, `QueryRangeRequest` is never initialised in `ProcessRequest`.  I don't even recall why its there.  We should probably remove it, and put this call into `queryRangeTerminator`.

Reply:
Thanks for the tip!